src/agent.cpp

```diff
@@ -249,62 +277,68 @@ void Agent::actionImpl() {
   }
 
   // For now let's not worry about turning the neck or setting the vision.
-  // However, do this now so doesn't override anything changed by the requested action.
-  // TODO for librcsc: setViewActionDefault, setNeckActionDefault that will not overwrite if already set.
+  // But do the settings now, so that doesn't override any set by the actions below.
+  // TODO: Add setViewActionDefault, setNeckActionDefault to librcsc that only set if not already set.
   this->setViewAction(new View_Tactical());
   this->setNeckAction(new Neck_TurnToBallOrScan());
 
+  const WorldModel & wm = this->world();
+  bool may_fix = wm.self().collidesWithPost();
```

Comment:
I was referring to the existing HFO codebase (not your code!).

I was proposing going back to the boolean return value: False indicates known failure and True indicates everything else (e.g. possible success/unknown). IMO this would be simpler and easier to understand while still enabling the scenarios you described.